web-common/src/features/dashboards/url-state/filters/expression.ne

```diff
@@ -0,0 +1,51 @@
+@preprocessor esmodule
+# match an expression used in url filters
+# adapted from https://dev.mysql.com/doc/refman/5.7/en/expressions.html
+@builtin "whitespace.ne"
+@builtin "number.ne"
+@builtin "string.ne"
+
+@{%
+  import {
+    binaryPostprocessor,
+    inPostprocessor,
+    havingPostprocessor,
+    andOrPostprocessor,
+  } from "./post-processors.ts";
+%}
+
+expr => boolean_expr                             {% id %}
+      | (boolean_expr _ "AND"i _):+ non_and_expr {% andOrPostprocessor %}
+      | (boolean_expr _ "OR"i _):+ non_or_expr   {% andOrPostprocessor %}
+
+# these are used to disambiguate matches
+non_and_expr => boolean_expr                            {% id %}
+              | (boolean_expr __ "OR"i __):+ non_and_expr {% andOrPostprocessor %}
+non_or_expr  => boolean_expr                            {% id %}
+              | (boolean_expr __ "AND"i __):+ non_or_expr {% andOrPostprocessor %}
```

Comment:
I'm not totally following these. At first glance, it looks like adding these conditions would prevent multiple "ANDs" and multiple "ORs" in the same statement. But I see in the test cases that "X and Y and Z" is possible. So, mind just clarifying in the comment what these do?

Reply:
These are used in `expr` `(boolean_expr _ "AND"i _):+ non_and_expr` for AND. It matches continuous AND ended by a non AND expr.

Added some comments around this.